protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```
@@ -0,0 +1,137 @@
+                _originator, 
+                _recipient, 
+                _hashLock, 
+                _timeLock, 
+                block.timestamp
+        ));
+
+        // Check if the bridge transfer already exists
+        require(bridgeTransfers[_bridgeTransferId].amount == 0, "Bridge transfer already exists");
+
+        bridgeTransfers[_bridgeTransferId] = BridgeTransfer({
+            amount: totalAmount,
+            originator: _originator,
+            recipient: _recipient,
+            hashLock: _hashLock,
+            timeLock: block.timestamp + _timeLock,
+            exists: true
+        });
+
+        emit BridgeTransferInitiated(_bridgeTransferId, _originator, _recipient, _hashLock, _timeLock);
+        return _bridgeTransferId;
+    }
+
+    function completeBridgeTransfer(bytes32 _bridgeTransferId, bytes32 _secret) external override {
+        BridgeTransfer storage bridgeTransfer = bridgeTransfers[_bridgeTransferId];
+        require(bridgeTransfer.exists, "Bridge transfer does not exist");
+        require(keccak256(abi.encodePacked(_secret)) == bridgeTransfer.hashLock, "Invalid secret");
+
+        // Move the bridge transfer to completed
+        completedBridgeTransfers[_bridgeTransferId] = bridgeTransfer;
+
+        // Delete from active bridgeTransfers
+        delete bridgeTransfers[_bridgeTransferId];
+
+        payable(bridgeTransfer.recipient).transfer(bridgeTransfer.amount);
+
+        emit BridgeTransferCompleted(_bridgeTransferId, _secret);
+    }
+
+    function refundBridgeTransfer(bytes32 _bridgeTransferId) external override {
+        BridgeTransfer storage bridgeTransfer = bridgeTransfers[_bridgeTransferId];
+        require(bridgeTransfer.exists, "Bridge transfer does not exist");
+        require(block.timestamp > bridgeTransfer.timeLock, "Timelock has not expired");
+
+        delete bridgeTransfers[_bridgeTransferId];
+
+        payable(bridgeTransfer.originator).transfer(bridgeTransfer.amount);
```

check for silent failure